agent/main.go

```diff
@@ -554,11 +556,34 @@ func (cfg *agentConfig) getConfigMap(name string) (*apiv1.ConfigMap, error) {
 	return cm, nil
 }
 
+var validResources = []string{"rds", "classic-elb"}
+
+func isValidResource(name string) bool {
+	for _, resource := range validResources {
+		if name == resource {
+			return true
+		}
+	}
+	return false
+}
+
 func parseCloudwatchYaml(cm string) (*cloudwatch, error) {
+
 	cw := cloudwatch{}
 	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewBufferString(cm), 1000).Decode(&cw); err != nil {
 		return nil, err
 	}
+
+	// Validate AWS resources types.
+	if len(cw.Resources) == 0 {
```

Comment:
nit: I would just pass along the `cw.Resources` to a separate function that does all these checks. More of a nit, rather then anything. But that way its cleaner and we can test with more cases.